ros/api/v1/recommendations.py

```diff
@@ -59,6 +59,17 @@ def get(self, host_id):
             abort(404, message="host with id {} doesn't exist"
                   .format(host_id))
 
+        # cloud_provider & instance_type are necessary for suggestions
+        if (
+                system.cloud_provider is None
+                or system.instance_type is None
+        ):
+            return {
+                'inventory_id': system.inventory_id,
```

Comment:
Will this work with frontend?

Reply:
It should, we do have scenarios where-in there are no suggestions w.r.t. a system.
@PreetiW Could you confirm?

Reply:
Yes there is. For optimized state.

Reply:
I don't know if this is correct to show others states having no cloud_provider or instance_type as `no suggestion`.

Reply:
@kgaikwad Could you re-confirm on this?

Reply:
I am not in favor of sending same result like optimized state when there is missing data. @saltgen  could you explain how frontend can make difference between these two situations?

Reply:
@kgaikwad Atm there should be no difference, if `cloud_provider` info is missing we are treating the system as non-ROS.
Any suggestions if there should be any changes on BE and/or FE?

Reply:
@kgaikwad In case we don't need this, shall I close this out?

Reply:
+1 for closing this. I don't have any other opinion.

Reply:
Closing this out along with the JIRA card